packages/cli/src/commands/eject.ts

```diff
@@ -12,18 +13,34 @@ export type EjectOptions = {
 
 export const handleEject = async ({ argv }: CommandArgs<EjectOptions>) => {
   process.stdout.write(`\nLaunching eject using NPX.\n\n`);
-  const npxExecutableName = process.platform === 'win32' ? 'npx.cmd' : 'npx';
-  spawn(
+  const isWindowsPlatform = process.platform === 'win32';
+
+  const npxExecutableName = isWindowsPlatform ? 'npx.cmd' : 'npx';
+  const path = isWindowsPlatform ? sanitizePath(argv.path ?? '') : argv.path ?? '';
+  const projectDir =
+    isWindowsPlatform && argv['project-dir']
+      ? sanitizePath(argv['project-dir'])
+      : argv['project-dir'];
```

Comment:
I find it hard to read. I wonder if it would be actually simpler to split this command execution for different platforms using `if`:

```ts
if (isWindowsPlatform) {
  ...
} else {
  ...
}
```

Also, please check if we can abstract the code for all the Realm wrapper commands (perhaps per platform), as I believe they might share a lot in common (just to avoid duplication/divergence).